pyuploadcare/dj/conf.py

```diff
@@ -24,25 +24,70 @@
 
 cdn_base = settings.UPLOADCARE.get("cdn_base")
 
+upload_base_url = settings.UPLOADCARE.get("upload_base_url")
 
-widget_version = settings.UPLOADCARE.get("widget_version", "3.x")
-widget_build = settings.UPLOADCARE.get(
-    "widget_build", settings.UPLOADCARE.get("widget_variant", "full.min")
-)
+legacy_widget = settings.UPLOADCARE.get("legacy_widget", False)
+
+use_hosted_assets = settings.UPLOADCARE.get("use_hosted_assets", True)
 
-widget_filename = "uploadcare.{0}.js".format(widget_build).replace("..", ".")
+# Legacy widget (uploadcare.js)
 
-hosted_url = "https://ucarecdn.com/libs/widget/{version}/{filename}".format(
-    version=widget_version, filename=widget_filename
+legacy_widget_version = settings.UPLOADCARE.get("legacy_widget_version", "3.x")
+legacy_widget_build = settings.UPLOADCARE.get(
+    "legacy_widget_build",
+    settings.UPLOADCARE.get("legacy_widget_variant", "full.min"),
+)
+legacy_widget_filename = "uploadcare.{0}.js".format(
+    legacy_widget_build
+).replace("..", ".")
+legacy_hosted_url = (
+    "https://ucarecdn.com/libs/widget/{version}/{filename}".format(
+        version=legacy_widget_version, filename=legacy_widget_filename
+    )
 )
 
-local_url = "uploadcare/{filename}".format(filename=widget_filename)
+legacy_local_url = "uploadcare/{filename}".format(
+    filename=legacy_widget_filename
+)
+legacy_uploadcare_js = (
+    legacy_hosted_url if use_hosted_assets else legacy_local_url
+)
 
-use_hosted_assets = settings.UPLOADCARE.get("use_hosted_assets", True)
+if "legacy_widget_url" in settings.UPLOADCARE:
+    legacy_uploadcare_js = settings.UPLOADCARE["legacy_widget_url"]
+
+# New widget (blocks.js)
+
+widget_options = settings.UPLOADCARE.get("widget_options", {})
+
+widget_version = settings.UPLOADCARE.get("widget_version", "0.26.0")
+widget_build = settings.UPLOADCARE.get(
+    "widget_build", settings.UPLOADCARE.get("widget_variant", "regular")
+)
```

Comment:
So many places to configure the version and a variant. :-(

Reply:
yup, maybe make the setting a dict?

Reply:
```python
UPLOADCARE = {
    "widget": {
        "version": 0.x,
        "variant": "regular",
        "sourcers": "url, etc",
    }
}
```

Reply:
@dmitry-mukhin I can change that. Should I do the same for the `legacy_widget_*` settings?

Also, at this point, I'd like to type `settings.UPLOADCARE` so that we can have a spec/list of all the available settings in one place (also, add it to the docs?).

Reply:
same for legacy, sure.
good thing with settings dicts is when it's done right, you can just do `Instance(**CONF)`

Reply:
I'm going this route (work in progress):

```python
class LegacyWidgetSettingsType(TypedDict):
    version: str
    build: str

class SettingsType(TypedDict):
    pub_key: str
    secret: str
    cdn_base: typing.Optional[str]
    upload_base_url: str
    use_legacy_widget: bool
    use_hosted_assets: bool  # TODO
    legacy_widget: LegacyWidgetSettingsType


DEFAULT_CONFIG: SettingsType = {
    "pub_key": "",
    "secret": "",
    "cdn_base": None,
    ...
}

config = deepcopy(DEFAULT_CONFIG)
deep_update(config, settings.UPLOADCARE)  # type: ignore
```

This is to eliminate most of the individual variables inside the `pyuploadcare.dj.conf` module while also providing a clear overview of all the settings available along with their default values. This module will become more concise and less cluttered.

@dmitry-mukhin

Reply:
@dmitry-mukhin updated the whole `pyuploadcare.dj.conf` module.